pumi/pumi_mesh.cc

```diff
@@ -437,12 +422,17 @@ void pumi_mesh_print (pMesh m, bool print_ent)
   int* global_local_entity_count = new int[4*m->getPCU()->Peers()]; 
   int* global_own_entity_count = new int[4*m->getPCU()->Peers()]; 
 
-  MPI_Allreduce(local_entity_count, global_local_entity_count, 4*m->getPCU()->Peers(), 
-                MPI_INT, MPI_SUM, m->getPCU()->GetMPIComm());
-
-  MPI_Allreduce(own_entity_count, global_own_entity_count, 4*m->getPCU()->Peers(), 
-                MPI_INT, MPI_SUM, m->getPCU()->GetMPIComm());
-
+  memcpy(
+    global_local_entity_count, local_entity_count,
+    sizeof(int) * 4 * m->getPCU()->Peers()
+  );
+  m->getPCU()->Add(global_local_entity_count, 4);
+
+  memcpy(
+    global_own_entity_count, own_entity_count,
+    sizeof(int) * 4 * m->getPCU()->Peers()
+  );
+  m->getPCU()->Add(global_own_entity_count, 4);
```

Comment:
I think this needs a comment describing what's going on here. Should there be a `PCU_Allreduce`?